app/javascript/legacy_react/src/lib/utils.spec.ts

```diff
@@ -0,0 +1,30 @@
+import 'jest';
+import * as Utils from './utils'
+
+
+describe('Utils.removeChar', () => {
+  var teststring = '$$3,,20$$,';
+  var teststring2 = 'this string'
+
+  test("remove $,", () => 
+    expect(Utils.removeChar(teststring,'$,')).toBe('320')
+  )
+
+  test("remove number", () => 
+    expect(Utils.removeChar(teststring,'3')).toBe('$$,,20$$,')
+  )
+
+  test("remove blank spaces", () => 
+    expect(Utils.removeChar(teststring2,' ')).toBe('thisstring')
+  )
+
+  test("remove occurrences of a char from a string", () => 
+    expect(Utils.removeChar(teststring2,'s')).toBe('thi tring')
+  )
+
+  test("does not remove if not on argument", () => 
+    expect(Utils.removeChar(teststring2,'abcdef')).toBe('this string')
+  )
+
```

Comment:
Are there any other conditions you might be forgetting?

If you look at the types of `i` and `charArray` you can receive a string or null or undefined. You should test for those situations and verify the code works as you expect.

Alternatively, should those function arguments accept all of those types?

Reply:
You're right the only type accepted should be string